src/common/gcode_filter.cpp

```diff
@@ -0,0 +1,89 @@
+#include "gcode_filter.hpp"
+
+#define RETURN_GCODE                                \
+    char *gcode = prepareGcode();                   \
+    if (gcode == NULL) {                            \
+        continue; /* Nothing to return, continue */ \
+    }                                               \
+    return gcode
+
+char *GCodeFilter::nextGcode(State *state) {
+    for (;;) {
+        char nextByte = getByte(state);
+        if (*state == Eof) {
+            return buffer;
+        }
+        if (*state != Ok) {
+            return NULL;
+        }
+
+        if (nextByte == '\n') {
```

Comment:
Not possible in this case (new line), as there is an `else` block. I can use switch for other cases, but this would mean using two different ways (switch and if) within the same function and also it would increase code indentation.